src/core/ledger/grainAllocation.js

```diff
@@ -48,13 +57,26 @@ export type Immediate = "IMMEDIATE";
  */
 export type Special = "SPECIAL";
 
-export type AllocationPolicy = RegularPolicy | SpecialPolicy;
+export type AllocationPolicy = RegularPolicy | RecentPolicy | SpecialPolicy;
+
+export type RegularPolicy = BalancedPolicy | ImmediatePolicy;
```

Comment:
In that case, we don't have a need for the `RegularPolicy` type anymore, right? Let's just have `AllocationPolicy = Immediate | Balanced | Recent | Special`